### PR TITLE
ci: setup CI using script

### PR DESCRIPTION
**Pull request by moriarty**
- Run the same script that packages in ros-controls github org use
- https://rtw.stoglrobotics.de/master/use-cases/ros_packages/configure_repository.html

These will likely fail due to serial not being released for ROS2
- see #21
- https://github.com/wjwwood/serial/issues/204
- https://github.com/wjwwood/serial/issues/283

I don't want to squash merge this PR

### CODE REVIEW

**Comment by moriarty** [comment]
Note this PR is entirely auto-generated. I'm expecting it to fail, and will fix it up accordingly

**Comment by moriarty** [comment]
@abake48 or @MarqRazz I don't want to squash merge this PR because I want to easily revert specific commits in the future assuming either serial is released or there is a pure cmake version that we can depend on

File: .github/workflows/ci-format.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# This is a format job. Pre-commit has a first-party GitHub action, so we use
+# that: https://github.com/pre-commit/action
+
+name: Format
+
+on:
+  workflow_dispatch:
+  pull_request:
```
Comment (MarqRazz) [inline .github/workflows/ci-format.yml:8]: Should we also add a test when pushed to main?
```
  push:
    branches:
      - main
```

File: robotiq_driver/src/robotiq_gripper_interface.cpp
Comment (MarqRazz) [inline robotiq_driver/src/robotiq_gripper_interface.cpp:1]: Lots of these changes are going to be modified by `pre-commit`. Are you planning on running it in the PR?

File: .github/workflows/README.md
Comment (MarqRazz) [inline .github/workflows/README.md:1]: Should we shorten this file to just three tables and the `Explanation of different build types` listed once?

**Review by MarqRazz — COMMENTED** [review]
Is the goal to have all test pass before we merge this PR?

File: robotiq_driver/src/robotiq_gripper_interface.cpp
Comment (moriarty) [reply robotiq_driver/src/robotiq_gripper_interface.cpp:1]: yes

File: .github/workflows/README.md
Comment (moriarty) [reply .github/workflows/README.md:1]: This README is inside the workflows directory and is the full list, the shortened list goes in the main top level README and has a link to the full list.

**Comment by moriarty** [comment]
> Is the goal to have all test pass before we merge this PR?

Yes... I didn't expect this PR to take so long.

File: .github/workflows/ci-format.yml (hunk 1 above)
Comment (moriarty) [inline .github/workflows/ci-format.yml:8]:
```suggestion
  pull_request:
  push:
    branches:
      - main
```

**Review by MarqRazz — APPROVED** [review]
Do we expect each ros release to need a different `.repos` file? Would be nice if we could squash that down to a single file. This PR adds so many configuration files (but a lot of tests too!).

**Comment by moriarty** [comment]
> Do we expect each ros release to need a different `.repos` file? Would be nice if we could squash that down to a single file. This PR adds so many configuration files (but a lot of tests too!).

Basically the reason for each repo to have a uniq repos file is that they could require different branches...

@destogl has a script which sets this all up, and is what is used for the ROS 2 Control related repos.

I do agree that it pollutes the top directory with a bunch of stuff so maybe I re-organize and move them into a subdir

**Pull request merged**